code/game/objects/objs.dm

```diff
@@ -48,6 +48,11 @@ var/global/list/reagents_to_log = list(FUEL, PLASMA, PACID, SACID, AMUTATIONTOXI
 	if(src in processing_objects)
 		processing_objects -= src
 
+	if(integratedpai)
+		integratedpai.forceMove(get_turf(src))
```

Comment:
So in the example case where the pAI is riding in a MULE and the MULE gets gunned down, the pAI would just die along with the MULE?
Not disagreeing, just confirming.